rct229/rulesets/ashrae9012019/section16/section16rule3.py

```diff
@@ -0,0 +1,99 @@
+    compare_schedules,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+
+class Section16Rule3(RuleDefinitionListIndexedBase):
+    """Rule 3 of ASHRAE 90.1-2019 Appendix G Section 16 (Elevators)"""
+
+    def __init__(self):
+        super(Section16Rule3, self).__init__(
+            rmds_used=produce_ruleset_model_instance(
+                USER=False,
+                BASELINE_0=True,
+                PROPOSED=True,
+            ),
+            each_rule=Section16Rule3.ElevatorRule(),
+            index_rmd="baseline",
+            id="16-3",
+            description="The elevator cab ventilation fan shall be modeled with the same schedule as the elevator motor.",
+            ruleset_section_title="Elevators",
+            standard_section="Section G3.1",
+            is_primary_rule=True,
+            rmd_context="ruleset_model_descriptions/0",
+            list_path="ruleset_model_descriptions[0].buildings[*].elevators[*]",
+            required_fields={"$": ["calendar"], "$.calendar": ["is_leap_year"]},
+            data_items={"is_leap_year_b": (BASELINE_0, "calendar/is_leap_year")},
+        )
+
+    def is_applicable(self, context, data=None):
+        rmd_p = context.PROPOSED
+
+        return (
+            len(
+                find_all(
+                    "$.ruleset_model_descriptions[0].buildings[*].elevators", rmd_p
+                )
+            )
+            > 0
+        )
+
+    class ElevatorRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section16Rule3.ElevatorRule, self).__init__(
+                rmds_used=produce_ruleset_model_instance(
+                    USER=False,
+                    BASELINE_0=True,
+                    PROPOSED=True,
+                ),
+            )
+
+        def get_calc_vals(self, context, data=None):
+            building_b = context.BASELINE_0
+            building_p = context.PROPOSED
+
+            is_leap_year_b = data["is_leap_year_b"]
+            mask_schedule = [1] * 8784 if is_leap_year_b else [1] * 8760
+
+            cab_ventilation_schedule_b = getattr_(
+                building_b,
+                "building",
+                "elevators",
+                "cab_ventilation_fan_multiplier_schedule",
+            )
+            cab_ventilation_schedule_p = getattr_(
+                building_p,
+                "building",
+                "elevators",
+                "cab_ventilation_fan_multiplier_schedule",
+            )
+
+            compare_schedules = compare_schedules(
+                cab_ventilation_schedule_b,
+                cab_ventilation_schedule_p,
+                mask_schedule,
+                is_leap_year_b,
+            )
+
+            return {
+                "cab_ventilation_schedule_b": cab_ventilation_schedule_b,
+                "cab_ventilation_schedule_p": cab_ventilation_schedule_p,
+                "compare_schedules": compare_schedules,
+            }
```

Comment:
Think the issue here is the same as the `16-2`

Reply:
Addressed